packages/rpc-methods/src/restricted/notify.test.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  getImplementation,
+  getValidatedParams,
+  NotificationType,
+} from './notify';
+
+describe('snap_notify', () => {
+  const validParams = {
+    type: 'inApp',
+    message: 'Some message',
+  };
+
+  describe('getImplementation', () => {
+    it('shows inApp notification', async () => {
+      const showNativeNotification = jest.fn().mockResolvedValueOnce(true);
+      const showInAppNotification = jest.fn().mockResolvedValueOnce(true);
+
+      const notificationImplementation = getImplementation({
+        showNativeNotification,
+        showInAppNotification,
+      });
+
+      await notificationImplementation({
+        context: {
+          origin: 'extension',
+        },
+        method: '',
+        params: {
+          type: NotificationType.inApp,
+          message: 'Some message',
+        },
+      });
+
+      expect(showInAppNotification).toHaveBeenCalledWith('extension', {
+        type: NotificationType.inApp,
+        message: 'Some message',
+      });
+    });
+
+    it('shows native notification', async () => {
+      const showNativeNotification = jest.fn().mockResolvedValueOnce(true);
+      const showInAppNotification = jest.fn().mockResolvedValueOnce(true);
+
+      const notificationImplementation = getImplementation({
+        showNativeNotification,
+        showInAppNotification,
+      });
+
+      await notificationImplementation({
+        context: {
+          origin: 'extension',
+        },
+        method: '',
+        params: {
+          type: NotificationType.native,
+          message: 'Some message',
+        },
+      });
+
+      expect(showNativeNotification).toHaveBeenCalledWith('extension', {
+        type: NotificationType.native,
+        message: 'Some message',
+      });
+    });
+
+    it('throws an error if the notification type is invalid', async () => {
+      const showNativeNotification = jest.fn().mockResolvedValueOnce(true);
+      const showInAppNotification = jest.fn().mockResolvedValueOnce(true);
+
+      const notificationImplementation = getImplementation({
+        showNativeNotification,
+        showInAppNotification,
+      });
+
+      await expect(
+        notificationImplementation({
+          context: {
+            origin: 'extension',
+          },
+          method: '',
```

Comment:
Please use the proper method name for all tests.

Reply:
Updated for all now.